### ISSUE TITLE
Emby Server (ex-Mediabrowser Server) [$10]

### ISSUE THREAD

**Issue by AlwinHummels** [SynoCommunity/spksrc#1346]
I like to have http://mediabrowser.tv/ server to run on my NAS. Is it possible for some developers here to take a look at it? I hope this could be realized ;-)

Thanks anyway.
##

**Comment by salman2learn** [comment]
@LukePulverenti 
I can help with C# wrapper code. 
Also, can you tell why this existing .Net wrapper won't work: http://magick.codeplex.com/

**Comment by LukePulverenti** [comment]
Excellent, thanks! That one will not work because it's a mixed mode assembly that will only function on windows. It might be possible to rip out the native assembly but it's really a large project. We only need about 10 basic functions. I will create a new repository that has the current code and then respond again.

**Comment by LukePulverenti** [comment]
Here it is https://github.com/MediaBrowser/ImageMagickSharp

Run the unit tests and you'll find two failing tests. Once those are finished there's about 5 other methods we'll need, but they're not that difficult.

**Comment by rvmourik** [comment]
Not to be impatient, but is there any progress? I see a lot of other media browser activity :-)

**Comment by n00b42** [comment]
HI, i'm also interested in this project. 
1) I'm quite familiar with C#, but as I never worked with github before (worked with git, but no branch/merge/...) and it might take some time to get into it, I would like to know how the progress is.
2) And how much work besides this library wrapper is needed to get MediaBrowser to run on ARMbased Synology's? As I might put some work in it is quite feasible to have MediaBrowser there in the next weeks, but I may not be interested if it's clear that it might need many more months.
3) Would it work on Syn DS414j (MINDSPEED Comcerto C2000)? Mono seems fine/available.

**Comment by LukePulverenti** [comment]
@n00b42 

Awesome, great to hear. After the ImageMagick library wrapper is done, we need to update MBS to use ImageMagick instead of GDI+. That part is easy. Then we have to start working on the Synology package. Perhaps that can get going in parallel?

We already have someone working on the ImageMagick wrapper and it's going quite well. It's getting close to complete. If Synology package building is not your forte then there are always other projects you could help on.

**Comment by n00b42** [comment]
@LukePulverenti 
Thanks for the informations, it is nice to hear that it's going quite well.
I have to admit that i'm not familiar with Synology package building (as packaging in Linux environments at all) ... I'll follow the issue and sure will help testing / come up with suggestions.
(Due to another project I might be short on time for the next month(s) )

**Comment by LukePulverenti** [comment]
Just a heads up, our ImageMagick library is complete and we will now begin replacing GDI with ImageMagick for Media Browser Server image processing operations. This will be done for the next server release cycle (not the upcoming beta that is about to go stable).

Once this is complete all systems should be go. If there is someone who would like to volunteer to simultaneously start working on a Synology package, that would be most appreciated.

**Comment by spaceboy76** [comment]
This is brilliant, hopefully someone can pick up the baton for the final leg!

**Comment by LukePulverenti** [comment]
Well this turned out to be very easy, and the integration is already done in the dev branch:

https://github.com/MediaBrowser/MediaBrowser/tree/dev

All traces of GDI+ have been removed so there should be no more blocking issues at this point.

**Comment by rvmourik** [comment]
Good to hear :-) Sadly i don't know how to build the package. But i'm sure someone here does.

**Comment by cytec** [comment]
yeah nice. well i think now everything _should_ work, i'll give it another try if i got some time left :)

@LukePulverenti as i don't have any ideas how to compile something with mono... is there a compiled version of your `dev` branch or do i need to compile it myself?

**Comment by LukePulverenti** [comment]
We have windows and linux build scripts available: https://github.com/MediaBrowser/MediaBrowser/tree/dev/Tools

**Comment by jabbera** [comment]
@cytec check out the way nzb drone package works. You shouldn't actually need to compile anything.

**Comment by AlwinHummels** [comment]
Look Like the imagemagick is to old 😕

**Comment by jabbera** [comment]
What do you mean? The existing package is too old? Upgrading it should hopefully be trivial, or adding another version if necessary.

**Comment by AlwinHummels** [comment]
See here my logs and what i'veel tried http://mediabrowser.tv/community/index.php?/topic/13890-synology-almost-working/page-2#entry180002

**Comment by LukePulverenti** [comment]
You should upgrade, and/or adjust the requirement in the ImageMagickSharp.dll.config file.

You can change that file to point to the version installed on your system. If the functions we're using are available in your older version, there's a good chance it will work.

**Comment by Dr-Bean** [comment]
To be clear, the whole point of the spksrc framework and SynoCommunity is to provide the software, with all it's dependencies in one self-contained package. In this case, it'll mean cross-compiling a supported version of ImageMagick. @cytec, I believe, already had a version running, so that could be a starting point.

Using an unmaintained and outdated IPKG package and adjusting MB to work with that is just about the opposite ;)

**Comment by cytec** [comment]
@jabbera i had a look at nzbdrone but still: i need to compile the dev branch to get an exe file that can be run by mono... 

@LukePulverenti thanks for the link, maybe i'll get that working... will try if i have a little time left

@AlwinHummels, @Dr-Bean jup. i've got ImageMagick working (latest version at that time was 6.9.0-0) so if there are no new depends (IM, Sqlite3, Mono) and once i (or someone else) can provide a compiled version of the dev branch, there shouldn't be a big problem getting it to work ;)

IF someone already has a mono ready compiled version of the dev branch available it would be great if that could be linked here so i can download it and try to make it work... i don't have much time lately and would be happy if i don't have to digg in that stuff myself :D

**Comment by rvmourik** [comment]
@cytec This link was posted on the MB forum by @LukePulverenti  for testing purposes:

https://www.dropbox.com/s/07hh1g4x9xo28jb/MBServer.Mono.zip?dl=0

This should be enough to run the tests.

Cheers

**Comment by cytec** [comment]
@rvmourik thanks

**Comment by cytec** [comment]
Good news! Got it running on my DS 1513+ (cedarview) 
still need to figure out some things like setting a different Data dir or a config file or something and need a lot of cleanup but... basically everything should work ;)

![bildschirmfoto 2015-02-23 um 14 44 13](https://cloud.githubusercontent.com/assets/1104903/6328717/83492bb0-bb6a-11e4-9df4-7d34035acbb3.png)

**Comment by rvmourik** [comment]
@cytec nice. I have a DS214+, i don't know how much work it is. But is it possible to provide a spk? This way i can test it for that cpu structure.

Good work!

**Comment by cytec** [comment]
@LukePulverenti is there a way to give MediaBrowser a specific config file or data directory on startup? currently everything get's saved in the same folder inside a ProgramData-Server folder (if i understand that correctly) due to updating mechanism and stuff it would be cool if all cache/user/config data would be stored in a different directory ... wasn't able to find any info about this besides changing it over the webinterface...

**Comment by GPDawes** [comment]
I have an older DS409 ( 	Marvell Kirkwood mv6281 ARM ) and a newer 112+ DS112+ 	(Marvell Kirkwood mv6282 2.0Ghz ARM (Marvell ARMADA 300)

i don't mind if its a command line and config files that are needed to start it/amend it. Can live without a GUI.

**Comment by cytec** [comment]
@rvmourik currently i have a spk for cedarview only and even that is still buggy and more or less just "quickly hacked together"... before i'll provide any testing stuff i'll make sure that everything is setup correctly and clean up the current mess... also i'd like to make sure that a proper Cache dir is set so one can update without loosing all of his data.

**Comment by rvmourik** [comment]
@cytec I get it :-) Hope @LukePulverenti can provide you with the right intel.

**Comment by cytec** [comment]
think i found it... seems like `-programdata` was the switch i was looking for ;)

**Comment by AlwinHummels** [comment]
@cytec What did you install / compile before it works? I can't get it work on my DS415+ it's so frustrating I'm doing the trial and error version of installing this. 
Did you change some files?

**Comment by cytec** [comment]
@AlwinHummels like @Dr-Bean mentioned before... cross compiling a lot of stuff... just useing old ipkg versions might not work... i compiled ImageMagick, sqlite3 and Mediainfo from scratch to make it work

**Comment by AlwinHummels** [comment]
@cytec ok I understand can you please tell me how to compile because that's where it goes wrong at mine.
I tried ./configure --prefix=/opt and then make this goes wrong
i tried ./configure and then make this goes also wrong 

thanks for all your help so far

**Comment by R-OG** [comment]
Nice work cytec, have you tried scanning your library yet. I have it working but no images are displayed in my library due to problems with image stick which i compiled from scratch.

**Comment by Dr-Bean** [comment]
@AlwinHummels Don't try compiling on the NAS, that's setting yourself up for failure ;) Follow the directions in the [readme](https://github.com/SynoCommunity/spksrc/blob/develop/README.rst) to set up a VM, then add a cross package to compile ImageMagick for the arch you need.

**Comment by cytec** [comment]
@R-OG yup, seems like it's working but i've only added my Music library for now... I'm going to compile some beta testing spk's right now.... i'll post here when they are finished ;)

**Comment by rvmourik** [comment]
@cytec Nice :-D

**Comment by AlwinHummels** [comment]
@cytec Nice indeed  :-D
I'm installing VMWare now and try to learn how to compile stuff myself ;-)

**Comment by LukePulverenti** [comment]
Available command line params are here:
http://mediabrowser.tv/community/index.php?/topic/3634-linux-setup/

**Comment by cytec** [comment]
@rvmourik spk for 214+ (armadaxp) should be online at http://cytec.us/spk/

cedarview should also be online right now...
@GPDawes  `88f6281` is currently compiling

**Comment by rvmourik** [comment]
@cytec Will try it later tonight. Thanks

**Comment by R-OG** [comment]
Hey cytec do you mind compiling for x86

**Comment by AlwinHummels** [comment]
Sorry no luck on my DS213 it stops MB and don't show anything.
Hope you can fix and make packages for avoton to ;-)
I'm get stuck on compiling etc...

**Comment by cytec** [comment]
@R-OG i'm currently compiling for all arch's... but MediaInfo takes ages to compile so it may take a while...

@AlwinHummels you should not start it from mono or anything... there is a script for that ;) you'll need a proper path setup which contains ImageMagick and other stuff! (thats the error you are getting btw) if you want to try to start it manually try calling `sh -x /var/packages/mediabrowser/scripts/start-stop-status start` which should give you some more output...

**Comment by LukePulverenti** [comment]
by the way take a look at the last link i provided, that documents the ffmpeg and ffprobe command line args. if unspecified the server will attempt to download static ffmpeg builds, but we haven't added arm to that yet. so i would suggest filling in both of those params.

**Comment by rvmourik** [comment]
@cytec Installed, it starts. But the Web Interface isn't reachable, after a while the package service stops. 

Maybe the problem is, what @LukePulverenti is saying in the previous post. If i am correct, the cpu in my ds214 is an ARM arch.

**Comment by cytec** [comment]
@LukePulverenti well... any special needed lib compiled in ffmpeg? 
afaik DSM ships with an `ffmpeg` binary but i dont think that there is an `ffprobe` binary also my `ffmpeg` spk currently doesn't support arm so i cant depend on that for now...

**Comment by LukePulverenti** [comment]
if you can match everything supplied here it should be ok: http://ffmpeg.gusari.org/static/

**Comment by cytec** [comment]
Damn... seems like i'll first have to go back trying to get ffpmeg/ffprobe working on arm devices before this will run :(

**Comment by rvmourik** [comment]
@cytec This package source that i use for another package also has a FFMpeg available for my DS:

http://packages.mdevries.org/

Don't know if it helps?

**Comment by cytec** [comment]
if there also is ffprobe available it may fix the problem. you could try by installing it and after that changing the start-stop script to point to the installed ffmpeg/ffprobe version...

**Comment by rvmourik** [comment]
Little update, the package i provided did not have ffprobe onboard. So i started thinking how does DS Video Station handle their video stuff. You can guess, it's ffmpeg and they also include ffprobe :-) So i edited the start stop script and mediabrowser started and i'm walking to the wizard as we speak :-) I'll keep you posted with the progress

**Comment by LukePulverenti** [comment]
Well done!

**Comment by AlwinHummels** [comment]
Grey work !!!

**Comment by cytec** [comment]
Nice seems to be new! i was pretty sure DSM only ships with ffmpeg... the more you know ;)

**Comment by cytec** [comment]
@rvmourik can you confirm that the `ffmpeg` and `ffprobe` binaries provided by VideoStation are working?

@Diaoul @Dr-Bean what do you think, can we depend on Synology's VideoStation for this SPK?
If we'll need a way around the `ff_get_cpu_flags_arm` failure which ffmpeg throws when trying to compile for armada/arm arches see: https://github.com/SynoCommunity/spksrc/issues/1446

**Comment by rvmourik** [comment]
@cytec It's still running, scanning of media went well as far as i can see. Haven't tried playing media files though. With a bit of luck i'll get to that toninght.

**Comment by cytec** [comment]
@rvmourik Nice! keep me updated ;)

**Comment by Dr-Bean** [comment]
@cytec, I'd rather see that we're able to provide ffmpeg and ffprobe ourselves, instead of depending on VideoStation. Of course, publishing a beta package with an dependency on VS is possible, but at the very least, let's make sure we've determined that #1446 doesn't have a quick fix (missing compile arg, flag or switch, a toolchain issue etc).
Personally, I've not had the opportunity to look into the ffmpeg issue more thoroughly than being able to reproduce it.

Obviously, for testing purposes and whatnot, great workaround.

**Comment by jabbera** [comment]
@cytec As an unsolicited opinion I would prefer to not need to install Video station. The fact that it compiles and runs under it means that it should be possible to do ourselves. (Plex does it also, but wrapped in their own code located here: http://files.plexapp.com/elan/ffmpeg/PlexNewTranscoder.tar.bz2)

**Comment by cytec** [comment]
@Dr-Bean yeah fixing that problem would be my prefered solution too. Currently testing a few compilation switches maybe one of them will fix it...

I know it MUST compile, somehow. Synology does it, and seems like others did it as well... so it must be something with switches or env settings... maybe i'll idle a little bit in #ffmpeg channel on freenode and see if someone there is able to help.

Maybe someone with a armada* DS can post the configuration switches here? @rvmourik would you be able to check with which options it was compiled for VS and post those here?

**Comment by rvmourik** [comment]
@cytec I want to do it, but i'm not sure how i can get that kind of information. I'm no Unix expert, but if you can point me in the right direction i can try to get the proper intel for you.

I can also try and contact the guys from mobile-harddisk.nl who compiled ffmpeg package with DTS support. Will try that anyway..

**Comment by cytec** [comment]
@rvmourik `ffmpeg -buildconf` should do the trick should output something like this:

    daenerys> ffmpeg -buildconf
    ffmpeg version 2.2.1 Copyright (c) 2000-2014 the FFmpeg developers
      built on Nov 11 2014 03:18:11 with gcc 4.2.0 (GCC)
      configuration: --extra-cflags=-I/home/cytec/spksrc/spk/ffmpeg/work-cedarview/install/usr/local/ffmpeg/include --extra-ldflags=-L/home/cytec/spksrc/spk/ffmpeg/work-cedarview/install/usr/local/ffmpeg/lib --disable-yasm --enable-  libfdk-aac --enable-gpl --enable-libx264 --enable-libspeex --enable-libvorbis --enable-libmp3lame --enable-nonfree --enable-cross-compile --arch=cedarview --target-os=linux --cross-prefix=/home/cytec/spksrc/toolchains/syno- cedarview/work/x86_64-linux-gnu/bin/x86_64-linux-gnu- --prefix=/usr/local/ffmpeg --extra-libs=-ldl
      libavutil      52. 66.100 / 52. 66.100
      libavcodec     55. 52.102 / 55. 52.102
      libavformat    55. 33.100 / 55. 33.100
      libavdevice    55. 10.100 / 55. 10.100
      libavfilter     4.  2.100 /  4.  2.100
      libswscale      2.  5.102 /  2.  5.102
      libswresample   0. 18.100 /  0. 18.100
      libpostproc    52.  3.100 / 52.  3.100
    
      configuration:
        --extra-cflags=-I/home/cytec/spksrc/spk/ffmpeg/work-cedarview/install/usr/local/ffmpeg/include
        --extra-ldflags=-L/home/cytec/spksrc/spk/ffmpeg/work-cedarview/install/usr/local/ffmpeg/lib
        --disable-yasm
        --enable-libfdk-aac
        --enable-gpl
        --enable-libx264
        --enable-libspeex
        --enable-libvorbis
        --enable-libmp3lame
        --enable-nonfree
        --enable-cross-compile
        --arch=cedarview
        --target-os=linux
        --cross-prefix=/home/cytec/spksrc/toolchains/syno-cedarview/work/x86_64-linux-gnu/bin/x86_64-linux-gnu-
        --prefix=/usr/local/ffmpeg
        --extra-libs=-ldl

**Comment by rvmourik** [comment]
@cytec if i run that command, i get the following:

```
DiskStation> ffmpeg -buildconf
ffmpeg version 2.0.2 Copyright (c) 2000-2013 the FFmpeg developers
  built on Jan  6 2015 23:29:01 with gcc 4.6.4 (Linaro GCC branch-4.6.4. Marvell GCC Dev 201310-2126.3d181f66 64K MAXPAGESIZE ALIGN)
  configuration: --prefix=/usr/syno --incdir='${prefix}/include/ffmpeg' --arch=arm --target-os=linux --cross-prefix=/usr/local/arm-marvell-linux-gnueabi/bin/arm-marvell-linux-gnueabi- --enable-cross-compile --enable-optimizations --enable-pic --enable-gpl --enable-shared --disable-static --enable-version3 --enable-nonfree --enable-libfaac --enable-encoders --enable-pthreads --disable-bzlib --disable-protocol=rtp --disable-muxer=image2 --disable-muxer=image2pipe --disable-swscale-alpha --disable-ffserver --disable-ffplay --disable-devices --disable-bzlib --disable-altivec --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libmp3lame --disable-decoder=amrnb --disable-encoder=zmbv --disable-encoder=dca --disable-encoder=ac3 --disable-encoder=ac3_fixed --disable-encoder=eac3 --disable-decoder=dca --disable-decoder=eac3 --disable-decoder=truehd --cc=/usr/local/arm-marvell-linux-gnueabi/bin/arm-marvell-linux-gnueabi-ccache-gcc --disable-doc
  libavutil      52. 38.100 / 52. 38.100
  libavcodec     55. 18.102 / 55. 18.102
  libavformat    55. 12.100 / 55. 12.100
  libavdevice    55.  3.100 / 55.  3.100
  libavfilter     3. 79.101 /  3. 79.101
  libswscale      2.  3.100 /  2.  3.100
  libswresample   0. 17.102 /  0. 17.102
  libpostproc    52.  3.100 / 52.  3.100
Unrecognized option 'buildconf'.
Error splitting the argument list: Option not found
```

I also contacted the repo owner on the Dutch Synology Forum

**Comment by cytec** [comment]
@rvmourik thanks, can you try if this one is working for you? 
http://cytec.us/files/ffmpeg_armadaxp-5.0_2.5.4-2.spk

**Comment by rvmourik** [comment]
@cytec package installed and changed the start stop script. Started MediaBrowser and it's running, so i think it likes the ffmpeg and ffprobe binary :-)

I'm not sure if got enough time to check if playback is working, also working at the moment. Is there something else you want me to test?

**Comment by cytec** [comment]
@rvmourik just if the binaries are callable. just wanna know if they are executable on your arch ;)

i don't think that they will work this way with MediaBrowser (as they currently are compiled without ANY libraries)

**Comment by rvmourik** [comment]
@cytec is it safe to assume that they are callable because mediabrowser is able to start while pointing to these binaries. Or do i need to execute some commands?

**Comment by rvmourik** [comment]
@cytec Playback doesn't work using the ffmpeg/ffprobe from VideoStation, it's missing some lib files as far as i can see.

**Comment by cytec** [comment]
Yup i expected that sadly -.-
I don't know if mediabrowser checks if the binaries are executable or bad ... Just run the same command like before (to show how it was compiled) should be enough to see if it compiled correctly :)

**Comment by cytec** [comment]
@rvmourik nevermind... just saw that there where a few librarys missing in the spk i posted...
can you try this version: http://cytec.us/files/ffmpeg_armadaxp_2.5.4-6.spk

just install and try to run `/usr/local/ffmpeg/bin/ffmpeg -version` that one will also NOT work for playback or anything else... just to check if it compiled/linked for armv7 successfully

**Comment by R-OG** [comment]
Hey cytec i just tried out the x86 spk on my DS3612xs but it say the package is not supported on my diskstation  platform.

**Comment by cytec** [comment]
@R-OG DS3612xs needs bromolow spk not x86 ;)

**Comment by R-OG** [comment]
Thanks cytec, I'm finally up and running and everything seems to be working perfectly.

**Comment by LukePulverenti** [comment]
Wow fantastic. Can we get the process documented? Thanks!

**Comment by AlwinHummels** [comment]
@R-OG that's great news can you please help me to make it also work for the arch avoton? (DS415+)
Non of the packages are there available jet, but I Like to learn how to make my own. I have a debian install and setup like written here: https://github.com/SynoCommunity/spksrc/blob/develop/README.rst
I tested the sonarr package and that's build fine but now the packages that aren't in the list how to compile those? like imagemagic and mediabrowser.server.mono

hope you can help me/us

**Comment by LukePulverenti** [comment]
Yea it would be great if someone could help us build a package that we can put on the website. 

Heads up, we host the mono zip here that you can always use: https://github.com/MediaBrowser/MediaBrowser.Releases/raw/master/Server/MediaBrowser.Mono.zip

But just be aware, that contains the latest stable release, which does not yet have ImageMagick.

**Comment by LukePulverenti** [comment]
And releases are also tagged in github: https://github.com/MediaBrowser/MediaBrowser/releases

**Comment by rvmourik** [comment]
@cytec will try to do it in the afternoon, otherwise i'll check the package tonight. Good to here that MB is working on the other archs with the working ffmpeg.

**Comment by spaceboy76** [comment]
tried the cedarview spk on my 2413+, but apparently it does not contain a digital signature?

**Comment by spaceboy76** [comment]
ok got round that

**Comment by spaceboy76** [comment]
its working, the instant thing i notice is there's no facility to restart MB3. the button next to shut down is not there and when i installed the server backup utility to restore over my existing MB3 config the button was missing from the notification that pops up telling you to restart the server. probably going to start to create threads on the MB3 forum now though... thanks for everyone's hard work here!

**Comment by Dr-Bean** [comment]
@spaceboy76 Note that this is an alpha version...not yet fit for 'production' purposes.

**Comment by cytec** [comment]
@LukePulverenti Im on it ;) just doesn't make any sense to publish something which is mostly unfinished. arm arches still aren't confirmed to work (due to missing ffmpeg) but i hope i get that sorted out once @rvmourik can confirm that he can run the provided binaries... then we/i just need to compile ffmpeg with some more libraries and everything should be good for cleaning up and making a beta SPK + pull request ;)

@AlwinHummels you have to make your own cross/ packages for those to compile, and also tweak a few settings. i reccomend you'll try out some stuff first like cross compiling etc... or just wait for it to get pushed out. I'm currently still useing an older version of spksrc and there is no avoton toolchain, i'll make a SPK for avoton once i have the needed dependencies converted to the newer spksrc version.

Basically the SPK's can also be installed for testing purpose if you'll add the url from https://github.com/SynoCommunity/spksrc/issues/1346#issuecomment-75585444 to you PackageCenter... but you'll have to turn of the security check as they are not signed in any way.

**Comment by cytec** [comment]
@AlwinHummels try that one: http://cytec.us/files/mediabrowser_avoton-5.1_20150225-3.spk

**Comment by AlwinHummels** [comment]
@cytec Thanks man it looks good but, when I install some plugins and restart the package starts but does nothing and after a short while it stops working

**Comment by cytec** [comment]
@AlwinHummels you know what ALPHA means do you?... 
i have not tested to install anything yet, just clean install of the server and indexing some music. I don't know how MB handles plugin installation and where they are saved or if they have any special depencencies etc. So if you'll do something like that, you are on your own... Im busy enough getting MB itself to work...

Maybe try to inspect the output of MB for whats wrong or run it with mono in debug mode (you'll need to change the start-stop script for that)...

**Comment by rvmourik** [comment]
@cytec 

```
/usr/local/ffmpeg/bin/ffmpeg -version
ffmpeg version 2.5.4 Copyright (c) 2000-2015 the FFmpeg developers
built on Feb 25 2015 00:30:13 with gcc 4.6.4 (Linaro GCC branch-4.6.4. Marvell GCC 201303-1655.6f19debc) 20120731 (prerelease)
configuration: --pkg-config=/usr/bin/pkg-config --enable-shared --disable-doc --disable-yasm --enable-gpl --enable-nonfree --enable-cross-compile --arch=generic --target-os=linux --cross-prefix=/home/cytec/spksrc/toolchains/syno-armadaxp/work/arm-marvell-linux-gnueabi/bin/arm-marvell-linux-gnueabi- --prefix=/usr/local/ffmpeg --extra-cflags=-I/home/cytec/spksrc/spk/ffmpeg/work-armadaxp/install/usr/local/ffmpeg/include --extra-ldflags=-L/home/cytec/spksrc/spk/ffmpeg/work-armadaxp/install/usr/local/ffmpeg/lib --extra-libs=' -ldl' --enable-libmp3lame --enable-libspeex --enable-libvorbis --enable-libx264 --enable-openssl --enable-libzmq
libavutil      54. 15.100 / 54. 15.100
libavcodec     56. 13.100 / 56. 13.100
libavformat    56. 15.102 / 56. 15.102
libavdevice    56.  3.100 / 56.  3.100
libavfilter     5.  2.103 /  5.  2.103
libswscale      3.  1.101 /  3.  1.101
libswresample   1.  1.100 /  1.  1.100
libpostproc    53.  3.100 / 53.  3.100
```

**Comment by cytec** [comment]
@rvmourik Yeah! seems like we got that! Now i can move to get the depends for ffmpeg to work :)

**Comment by rvmourik** [comment]
@cytec Nice :-) Maybe it's wise for me to wait for an updated ffmpeg version so i can test the playback properly. No need to test is when it's not 100% compiled, right?

**Comment by cytec** [comment]
@rvmourik if it's not to much work for you a playback test would be great... i don't know exactly which libraries are needed by MB for playback, the most common ones are compiled in that binary you are useing... maybe thats enough already ;)

**Comment by rvmourik** [comment]
@cytec Of course not, working on a common goal here! Will do it tonight, and come back to you :-)

**Comment by AlwinHummels** [comment]
@cytec It was no offence to you but thought I let you know and yes I know this is a very early "before" ALPHA ;-) I appreciate all your hard work and help.

**Comment by spaceboy76** [comment]
playback working here on DS2413+

**Comment by cytec** [comment]
@rvmourik i've updated the ffmpeg spk and included all libraries except vpx/webm and openjpeg which are enabled in the static build mentioned by @LukePulverenti so hopefully playback will work with this version :)

http://cytec.us/files/ffmpeg_armadaxp-5.0_2.5.4-8.spk

**Comment by rvmourik** [comment]
@cytec playback works in Kodi with MB extension (tested Movie/Serie MKV-files). Web player (Chrome) is stuttering, but this is an MB issue, has nothing to do with ffmpeg. Haven't had time to test DLNA.

**Comment by cytec** [comment]
@rvmourik Nice! did you use *-8.spk?

**Comment by rvmourik** [comment]
Yes i used the latest one

**Comment by rvmourik** [comment]
@cytec What is going to be the timeline? Is it safe to assume a Beta package can be released? That way we can test MB extensively and report the bugs on their forum.

**Comment by cytec** [comment]
@rvmourik not really... seems like ffmpeg compiles but like mentioned here: https://github.com/SynoCommunity/spksrc/issues/1446 it has massive performance issues due to missing optimization :/  

I'd like to have this sorted out before doing anything more "public" testing stuff... as there are a few users which assume BETA spk's to work correctly (also on arm devices)

also i'm currently haveing some difficulties compiling the newly created ffmpeg version for other arches. seems like some dependencies for libass need some finetuneing...

**Comment by rvmourik** [comment]
@cytec okay, i get it :-) Just read the other topic and that could explain the issues i'm encountering with the playback. If ffmpeg uses up almost every resource available it's logical that i expirience stuttering right?

**Comment by cytec** [comment]
@rvmourik yup, sadly it seems to be a bigger task to get that up and running properly :/

**Comment by rvmourik** [comment]
@cytec okay, that's a pitty. I am afraid i'm not the person to help with that kind of problems. Not if you want them fixed by the year 2022 :-)

If there is progress and you need me to test newly compiled binaries let me know, always willing to help.

**Comment by LukePulverenti** [comment]
Is there someone here who would like to run point in regards to building a synology package? We have a mailing list for all of our package maintainers for different platforms, so I'd like to get someone on that.

**Comment by cytec** [comment]
@LukePulverenti currently there are still some problems compiling ffmpeg for arm devices. Personally i'd like to fix that before moving on with this so that every platform has the same dependencies (speeking of ffmpeg spk) instead of just making it available for people who run it due to the download of static binaries on first run.

If that issue is solved makeing an SPK for MB isn't that much of an issue. Actually i have done most of it already. Just needs some cleanups and URL corrections once MB with ImageMagick is out of the dev branch ;)

**Comment by rvmourik** [comment]
Hi @cytec, i'm curious if you made any progress in figuring out what's causing the ffmpeg troubles on the ARM archs.

Thanks :-)

**Comment by cytec** [comment]
@rvmourik yup seems to be a problem with spksrc's build method... don't know why exactly... basically i'm waiting for @Dr-Bean or @Diaoul to check that :)

**Comment by jabbera** [comment]
While the arm ffmpeg issue is being worked out dsm 5.2 includes docker support and I belive there is a docker package for media browser.

https://www.synology.com/en-us/dsm/5.2beta/productivity

**Comment by LukePulverenti** [comment]
that is correct but our docker packages rely on our own static builds of ffmpeg and we currently don't have arm. so that's not going to work.

**Comment by jabbera** [comment]
docker only appears to be supported on the Intel synologys.

**Comment by cytec** [comment]
@rvmourik @LukePulverenti seems like the problem with ffmpeg compilation is now fixed, thanks to @Dr-Bean... so basically we are good to go

**Comment by rvmourik** [comment]
@cytec Nice, good to hear @Dr-Bean found the solution. If there's a new package i will test it for you.

**Comment by n00b42** [comment]
@cytec Is it possible to compile it for "comcerto2k" (DS414j with Mindspeed Comcerto 2000) as well? would like to test it as well (and help find/purge bugs). Thx.

**Comment by cytec** [comment]
@rvmourik @n00b42 there are just some additional clfags left to improve ffmpeg's performance on arm devices... keep an eye on https://github.com/SynoCommunity/spksrc/issues/1446 for progress ;) Once that is sorted out there should be nothing in the way to get this finally working on most devices ;)

**Comment by Dr-Bean** [comment]
@cytec, for now, the good thing is that we're able to compile ffmpeg for all arches without having to resort to binaries etc.
The large performance issues of armada375 are probably limited to that arch because of those flags. I have a couple other ideas to improve performance, but it's just going to take time and testing.

Imo, let's go ahead with packaging MediaBrowser: if nothing else, non-arm users are able to test the package and provide feedback.

**Comment by cytec** [comment]
@Dr-Bean well i'll try to build some binaries with your patch and my additional libraries as soon as i got some time for that, which hopefully will be later today or tomorrow

i'll keep you updated on the progress

**Comment by AlwinHummels** [comment]
I'm happy to see the progress off this but I've one question is there a package for the avaton to? for now I can't find this on the list and I'm happy to test. For now I have a pre Alpha and it seems to work whn I dont use cover art. and streaming from chanels.

**Comment by LukePulverenti** [comment]
The new server release is now in beta status. Look for stable within the next several days.

**Comment by spaceboy76** [comment]
Where can we pick up the beta release from? Top work everyone!

**Comment by Dr-Bean** [comment]
To be clear: there is no beta release of the Synology package.
There's a beta build of MediaBrowser, but that doesn't make it a package. That'll be a while yet, there are several things that need to be done. 
Please don't continue asking for new releases or packages (@spaceboy76, nothing personal, multiple people have been asking throughout this thread, you just happen to be the last one asking ;))

**Comment by jabbera** [comment]
If you have dsm 5.2 and an Intel nas that is docker capable you can run it now.

**Comment by spaceboy76** [comment]
;) i'm already running it! i'm was planning on rolling out to a few remote users this weekend but if there's going to be a stable package soon then i'll probably just wait for that.

**Comment by LukePulverenti** [comment]
By the way as far as package building goes, if you didn't see our announcement of our new project name, please see that here:

http://mediabrowser.tv/community/index.php?/blog/1/entry-179-introducing-emby/

On the mailing list with the linux package developers, we've decided to name the packages Emby Server. So i would suggest going with that.

**Comment by LukePulverenti** [comment]
A revision to the beta has been tagged, 3.0.5557.20000. I'd like to get the stable build out in a day or so.

**Comment by angel1835** [comment]
Version 3.0.5557.20000 works quite well on a DS214play. 
Streaming media to Web-CLient, DLNA and FireTV (XBMC) works very well. Good job and thx for that release.
Some issues with DLNA but don't matter for me.
Playback from channels failes with FFMPEG error 134 but this is also a known issue.

**Comment by spaceboy76** [comment]
realised this weekend that synology is not working for me. its fine at playing media direct within the same network but as soon as i tried to transcode a 720 video for remote playing on a roku it was pushing the synology to and past its limit. not a problem this has just clarified that i need a meaty windows pc to act at the main server which i am starting to build today i7 4790k ftw. good luck with this project and hope it works for those that continue!

**Comment by Diaoul** [comment]
@spaceboy76 : on your DS2413+, right?

**Comment by LukePulverenti** [comment]
The stable release with ImageMagick is now available:

https://github.com/MediaBrowser/MediaBrowser.Releases/raw/master/Server/MediaBrowser-3.0.5569.0.zip

Or this link without a version number:

https://github.com/MediaBrowser/MediaBrowser.Releases/raw/master/Server/MediaBrowser.Mono.zip

That second url will always point to the latest stable zip release.

**Comment by angel1835** [comment]
@LukePulverenti 

Playback from channels is working now.

Great job

**Comment by rvmourik** [comment]
@cytec Did you manage to compile a better version for ARM? Or are we waiting for changes at ffmpeg?

Thanks.

**Comment by cytec** [comment]
@rvmourik i think ffmpeg concerned errors are fixed, but had a lot to do in the last weeks hopefully i'll be able to take a closer look this week

**Comment by rvmourik** [comment]
@cytec That's some good news :-)

**Comment by cytec** [comment]
@LukePulverenti just a small question... are any binaries (imagemagick and/or mediainfo) needed for emby to run or are you useing the libraries directly?

**Comment by LukePulverenti** [comment]
we use imagemagick libraries and ffmpeg binaries. ffmpeg is the only dependency that we use binaries for.

**Comment by FlyGuy94** [comment]
@cytec Any chance we will see an emby package for synology on your website anytime soon?

**Comment by cytec** [comment]
nope, sadly not... 
still unable to stream with compiled ffmpeg seems like `libvpx` is needed which doesn't compile currently -.-
Also i'm getting a lot of errors for unknown commands passed to ffmpeg `-speed` and `-quality` for example @LukePulverenti can you help on that?

**Comment by LukePulverenti** [comment]
speed and quality are used by libvpx so if libvpx is absent I imagine that is why you'll see errors.

**Comment by cytec** [comment]
i thought i'll give you a little update on this one:

1. I was able to compile `ffmpeg` with all neded libraries and `yasm` support for my 1513+ (x64)
2. I created a PR for ImageMagick (which is also needed for emby to run)
3. I created a PR for emby itself (which depends on `ffmpeg` `ImageMagick` and `Mediainfo` spk)

so... it looks pretty good for now :)
next step is to compile stuff for other arches and see if that works as well (hopefully) and then wait for the PR`s to be merged and the Packages to be published (which can take some time again)

if someone wants to test the whole thing:

1. Install Mediainfo from SynoCommunity
2. Install ImageMagick from my Repo
3. Download and Install ffmpeg from here: http://cytec.us/syno/test-spk/ffmpeg/
4. Download and Install emby from here: http://cytec.us/syno/test-spk/emby/

hopefully that will work :)

NOTE: atm there are only `x64` builds available for `ffmpeg` and `emby` but i'm currently compiling for some other arches too (which takes quite a long time for ffmpeg) which should pop up as soon as they are done :)

**Comment by cytec** [comment]
@GPDawes @n00b42 `88f6281` and `comcerto2k` are also available now... any feedback is appreciated

**Comment by simono5** [comment]
Thank you @cytec. When I go to install Emby it's asking me to install Mono first. Is this expected?  I ask as it's not part of your install steps. Cheers

**Comment by cytec** [comment]
@simono5 yeah forget to mention that, fixed the install steps above... of course mono from SynoCommunity is also needed for emby to run :)

**Comment by Vlaves** [comment]
@cytec 
Thank you so much for this update :) I have one question. I know all here are against using the ffmpeg provided by synology's Videostation. But on a DS214play it would be great if I could use their version as it uses the hardware acceleration built into the Diskstation.

Is there any chance to incorporate their changes into your ffmpeg or maybe make emby use their ffmpeg if videostation is already installed. Then I would be able to use transcoding also for Live TV for my FireTVs :)

I would love to hear some positive feedback. And thanks again for all your hard work :)

Regards
Vlaves

**Comment by cytec** [comment]
@Vlaves that's not possible (see https://github.com/SynoCommunity/spksrc/issues/1346#issuecomment-75843089 playback is not working)
sadly the `ffmpeg` binaries provided by VideoStation don't support vpx/webm and therefore you won't be able to stream/transcode your media at all...

also like @Dr-Bean mentioned somewhere around here, the prefered solution would be to provide our own `ffmpeg` package instead of depending on a Synology one we don't have any influence.

so i don't think that will work... you __can__ test it of course. just need to edit `/var/packages/emby/scripts/start-stop-status` and change `FFMPEG="/usr/local/ffmpeg/bin"` to point at the VS path... also keep in mind that __afaik__ VideoStation doesn't support DTS

**Comment by LukePulverenti** [comment]
We only use vpx when needed, h264 is preferred. But chrome will sometimes use vpx, as well as older android devices that either don't support HLS or don't support it very well.

**Comment by simono5** [comment]
Good work, thank you.

Is it easy to suppress the @eaDir directory from showing up in media library's.

**Comment by bbollech** [comment]
I tested the x64 install, and so far it is working great for me.

**Comment by Vlaves** [comment]
@cytec Sadly I'm not able to install. Cause the ffmpeg package for evansport is not installed, although I anyway wanted to use the ffmpeg from VideoStation.
@LukePulverenti Thanks for your support here. If Emby is able to use the ffmpeg from synology, where can I set the parameters to use. Cause without the right parameters Hardware acceleration is not used. Is there some configfile where I can set the ffmpeg parameter that emby will use?

Thanks again to all for your support here. Regards
Vlaves

**Comment by LukePulverenti** [comment]
See here for command line args:

http://emby.media/community/index.php?/topic/3634-linux-setup/

But why not just start simple by building a functional package with a controlled ffmpeg?

**Comment by Vlaves** [comment]
@LukePulverenti Thnks for the reply. But I ment the command line parameters emby server is using when it calls ffmpeg for transcoding. There I need to add the parameters for using hardware acceleration.

Hope it is not hardcoded and can be adjusted in some configfile.

Regards
Vlaves

**Comment by LukePulverenti** [comment]
You can look in the Api project, then in the Playback folder. 

It would be great if we could just start with a releasable package for the masses before moving onto this though. Since the built in ffmpeg doesn't support vpx it's only going to be feasible for users who accept what that means. That's not good for casual joe user though because they're just going to be left wondering why things aren't working.

**Comment by Vlaves** [comment]
Thanks for the Info reg. the API Project.
Yes I would also like to have a releaseable package. Best would be to have that hardware acceleration part incorporated into the builtin ffmpeg.
A guy over here http://forum.serviio.org/viewtopic.php?f=14&t=16921&start=20 already incorporated the bits into "their" ffmpeg, so it should be doable. What do you think?

Thanks for your support.

**Comment by LukePulverenti** [comment]
Yes I think that's a great idea and it would make a lot of people happy.

**Comment by Vlaves** [comment]
Hope someone will be able to do that. Maybe you can contact this guy as he will probably sharing his ffmpeg source.

**Comment by demon1300** [comment]
cytec, any chance of getting an ImageMagick package for x64?  I'd love to do some testing on my DS1511+

**Comment by cytec** [comment]
@demon1300 yup: http://cytec.us/syno/test-spk/imagemagick/

@Vlaves im currently reading some stuff regarding hardware acceleration for evansport and ffmpeg... don't know if and how that will work but in the mean time you can install the evansport spk from here: https://cytec.us/syno/test-spk/ffmpeg/ and if you prefer the one provided by VideoStation, go ahead and change the start script like i mentioned before ;)

**Comment by cytec** [comment]
Also i think the guys from Plex did something like this in their PlexMediaTranscoder but it seems like both are very hacky and changed a lot of code to make it work basically -.-

**Comment by demon1300** [comment]
@cytec thanks!!  Working perfectly so far!

**Comment by rvmourik** [comment]
@cytec if i have time this week, i'll test the armadaxp build for his speed.

**Comment by Vlaves** [comment]
@cytec Thanks for the evansport ffmpeg build, so I could at least install it now. Thanks a lot for that :) Would be great if you could incorporate that hardware acceleration into your ffmpeg :)
Together with the tvheadend and emby, my Fire TV's could then display all my local media within the emby App (sorry dreaming a bit) :)

@all Regarding the @eaDir problem. Is there any way to hide it?

**Comment by cytec** [comment]
@Vlaves seems like a hughe chunk of work from what i can tell by taking a look at the synology ffmpeg source code... there is a __lot__ of stuff added and changed in order to make hardware acceleration work :( maybe someone else has more luck on this :/

About the @eaDir stuff... i don't know if there is an option to hide stuff/folders somewhere in the Settings or if @LukePulverenti has to add that to the code itself... but anyway that shouldn't be much of an issue

**Comment by bbollech** [comment]
@cytec Is there any difference between the "test-spk" Imagemagick and the one here https://cytec.us/spk/.  I used the bromolow Imagemagick package, and then the "test-spk" x64 package for Emby and ffmpeg.

**Comment by cytec** [comment]
@bbollech nope no difference just different arches... "test-spk" are compiled with x64 toolchain the ones in the repo are compiled for each arch itself

**Comment by choco55** [comment]
I just tested on a DS414 with this procedure (and the armadaxp packages) : 

Install Mono from SynoCommunity
Install Mediainfo from SynoCommunity
Install ImageMagick from my Repo https://cytec.us/spk/
Download and Install ffmpeg from here: http://cytec.us/syno/test-spk/ffmpeg/
Download and Install emby from here: http://cytec.us/syno/test-spk/emby/"

The dashboard is fast and the server is runing fine, but the library scan don't find any videos in the media folders I have defined.

**Comment by Vlaves** [comment]
@choco55 as far as I know, the users group needs to have at least read rights on the particular share. btw. under which user the emby server is operating?

**Comment by cytec** [comment]
@Vlaves emby runs under his own user (emby) which is in the users group... like every other spk does ;)

**Comment by Diaoul** [comment]
@cytec: maybe we can extract the patches to ffmpeg that Synology uses and use it too?

**Comment by choco55** [comment]
@Vlaves I activated read rights for the users group but I still have no movies after the libray scan

**Comment by LukePulverenti** [comment]
is the movies folder you added a network share? or are there symlinks involved?

**Comment by choco55** [comment]
the folders and movies are store on the NAS

**Comment by bbollech** [comment]
@choco55 You should be able to browse to the files in the dashboard when specifying the location.  If you can't see the videos when browsing, then you don't have the permissions set properly.

Once you add the local links, make sure to add path substitution links if you plan to perform any local playback.  The path substitution settings will allow for direct streaming, so it won't transcode if the local device supports the codecs from the video you are playing.

**Comment by choco55** [comment]
You are right, I didn't change the good setting for the users read access. It's ok now, the library scan in progress

**Comment by rvmourik** [comment]
@cytec Installed all the packages went well, setting up Emby configuration also works. 

All tests are with mkv files.

- Playing media with Kodi Emby Add-on (with DirectPlay) works like a charm.
- Playing from IOS client starts transcoding, CPU spikes to >95%, but the playback doesn't start

10mpbs quality:
- Playing from Chrome Web client starts transcoding, CPU goes to +/- 65%, but playback doesn't start
- Playing from Safari web client, CPU spikes to > 95%, playback starts and works good.

Ram goes to 88% for all tests.

@LukePulverenti when trying to start a movie from Safari it started a transcoding job, when i cancelled the playback from web client and starting a new playback (for instance a tv show) the transcoding log for this tv show still shows the filename from the movie transcoding job.

Also after stopping al playback, CPU usage is still at 95% and memory at 85%. So i looked at the process list and there were still three ffmpeg processes running at full speed. So it looks like transcoding isn't canceled, is this expected behaviour?

So i think the package it self is fine, but i think my cpu is to slow in some cases or it needs the hardware acceleration :-)

**Comment by bbollech** [comment]
@rvmourik What CPU do you have?  You can look in the transcoding logs and see how many fps it is able to transcode at.  Obviously your CPU needs to do at least 24fps or more depending on the framerate of the video.

As for the transcoding processes continuing to run, I know others have encountered that issue.  How are you stopping the movies?  I always click the X (inside the browser, not the browser itself) to close the play window, and I never encounter that issue.  I know in the past that some people would click pause and then browse backward and they would have the problem.

**Comment by rvmourik** [comment]
@bbollech armadaxp (Dual Core 1.33 GHz)

Tonight i'll look in the transcoding logs.

**Comment by Vlaves** [comment]
@LukePulverenti I have one question, which is not clear to me. What path substiturion settings to use to avoid transcoding with the Fire TV App. Is the Fire TV App supporting smb://...-Paths? I'm lil confused about this, as it is displaying everything in my Library, but it is almost always transcoding, which obviously takes to long on the DS214play without Hardware Acceleration...

Thanks for any Info :)

Btw. didn't have time to look into the API Project folder, so I could test with synology version of ffmpeg.

Regards
Vlaves

**Comment by n00b42** [comment]
Before I start: Thanks for all your work!

@LukePulverenti @Vlaves More or less the same for me: The DS414j is not well suited for transcoding. FireTV App lead to transcoding, while Kodi+Addon (PC) worked fine without transcoding.

@LukePulverenti @cytec I fear that the DS414j may be not powerful enough in the end... The web frontend works quite well around 90% of the time, sometimes it does not react and I have to reload. I tested the server with:
- PC: Emby Theater: lead to transcoding => started, but too slow, will test again.
While the Theater UI itself felt ultra responsive, I regularly got error messages, it crashed at least once, and the images took quite long to load (does Theater cache or reload after restart?)
- PC: Kodi+Emby Addon: worked quite well, sync of library worked, was able to see some videos without them beeing transcoded.
- FireTV: Emby for FireTV: interface feels laggy, long waiting for images, play lead to transcoding => playback started, but way to slow to watch (like alternating 2s playback, 3s buffering), will test again
- FireTV: Kodi+Emby: installed, library sync worked well, playback didnt seem to start, will test more intensive later.
- AndroidTablet: Emby for Android: UI feels more or less ok. As expected lead to transcoding => too slow to watch.

LOGS: Will investigate logs later: So far, I found really many exceptions due to failed HTTP requests (for images as far as I understand), one single case of an Out of memory situation, ...

**Comment by cytec** [comment]
@LukePulverenti I don't know if there is any option to disable transcoding in the settings at all (for slow devices like arm) so that `arm` users are at least able to install `emby` and use the server for clients which don't require transcoding?

@Diaoul from what i've seen they don't provide any patches in their source code, but they provide the whole changed source code for 2.2.x (which is quite old) and it looks like there was a hell of stuff changed to make hardware transcoding work properly :/ maybe we can contact Synology and ask if they have any patches we can use? 

@Vlaves @n00b42 @rvmourik i think transcoding isn't really a good idea to use on `arm` devices at all... Plex for example disabled transcoding on `arm` and `ppc` devices

From their website (https://support.plex.tv/hc/en-us/articles/201373823):

> Some NAS devices – primarily those with processors based on the ARM architecture – have capabilities low enough that transcoding is simply disabled altogether for them in Plex Media Server.

**Comment by cytec** [comment]
@Dr-Bean @Diaoul on that arm stuff... maybe there can be done a little bit of improvement with some more optimizations? Personally i don't have access to any arm based NAS at all and i don't think that any of them would be able to transcode properly no matter which flags/optimizations are done :(

@Vlaves you said VideoStation uses hardware transcoding? does that work for you? like can you play transcoded 1080p mkv files on your devices?

Also from what i've testet so far (720p 1Mbps):

* Chrome: is a little bit buggy... seems like `vpx` is a little slow when it comes to transcoding. cpu: ~85%
* Safari: works pretty well, buffering for like 10secs before playback starts and then plays smoothly- cpu: ~60%
* iOS: minimal buffering before playback starts (like 2secs) and then plays smoothly cpu: ~50%

stream mapping for Chrome:

  Stream #0:0 -> #0:0 (h264 (native) -> vp8 (libvpx))
  Stream #0:1 -> #0:1 (ac3 (native) -> vorbis (libvorbis))

and for iOS/Safari:

  Stream #0:0 -> #0:0 (h264 (native) -> h264 (libx264))
  Stream #0:1 -> #0:1 (ac3 (native) -> aac (native))

@LukePulverenti why are you useing libvpx/libvorbis for chrome? afaik Plex uses h264/aac for chrome as well

**Comment by rvmourik** [comment]
@cytec Tonight i will also test if VideoStation on my iPad works with transcoding (I believe it does).

**Comment by Diaoul** [comment]
@cytec: I dropped to my contact at Synology

**Comment by Vlaves** [comment]
@cytec yes, useing videostation I can perfectly play transcoded 1080p files on my Fire TV. And DVBLink is also using hardware transcoding with their ffmpeg version, don't know which version they use...

**Comment by cytec** [comment]
@Vlaves ok, lets wait for @Diaoul's contact at Synolgy to reply

**Comment by bbollech** [comment]
@Vlaves My understanding is that UNC paths should work fine for path substitution with the FireTV clients.

@n00b42 Your path substitution settings are either missing or incorrect, because Emby Theater should not transcode if played from the same LAN.

For those with transcoding issues, it is possible to place a duplicate MKV file in the same folder with codecs supported by the player being used, so that transcoding is avoided.  You can also use folder sync or cloud sync for the same purpose.  Those features will transcode the movies in advance to play later.

**Comment by n00b42** [comment]
As mentioned the log shows dozens of error messages:
- Error - App: Error getting response from
- Error - App: Error in The Open Movie Database
- Error - App: Error in TheMovieDb
- Error - MediaEncoder: Error sending q command to process  
Error - MediaEncoder: Error in WaitForExit  
Error - MediaEncoder: Error killing process
- Error - MediaEncoder: Error killing process
- Error - App: Error in ffprobe
- Error - ProviderManager: MovieDbEpisodeImageProvider failed in GetImageInfos for type Episode

Are they critical or unusual, or is it "normal" (at least the http request errors look like that might happen regularly) ?
( Reduced Log can be found here: https://paste.digitalworksempire.de/pohsgytfw/n7ef0o )

**Comment by bbollech** [comment]
@n00b42 Those app errors look like issues with the metadata fetcher trying to communicate with the fetching sites.  Can you access those sites manually?

I am not sure what the MediaEncoder errors would be.  I suggest going to the Emby forum and post your issues there to get better visibility.  There are people there that are way more qualified than me.  Otherwise maybe @LukePulverenti will reply.

**Comment by rvmourik** [comment]
This afternoon Emby crashed, this is the last info from the log before it crashed:

```
2015-05-06 12:14:38.9291 Info - App: Scan media library Completed after 0 minute(s) and 34 seconds
2015-05-06 12:14:38.9403 Info - LibraryMonitor: Watching directory /usr/local/emby/var/root
2015-05-06 12:14:38.9403 Info - LibraryMonitor: Watching directory /usr/local/emby/var/data/collections
2015-05-06 12:14:38.9412 Info - LibraryMonitor: Watching directory /usr/local/emby/var/data/playlists
2015-05-06 12:14:38.9412 Info - LibraryMonitor: Watching directory /usr/local/emby/var/data/camerauploads
2015-05-06 12:14:40.3602 Info - LibraryMonitor: Watching directory /volume1/videos/series
2015-05-06 12:14:40.5910 Info - LibraryMonitor: Watching directory /volume1/videos/movies
2015-05-06 12:49:17.5837 Error - Main: UnhandledException
	*** Error Report ***
	Version: 3.0.5597.1
	Command line: /usr/local/emby/share/emby/MediaBrowser.Server.Mono.exe -programdata /usr/local/emby/var -ffmpeg /usr/local/ffmpeg/bin/ffmpeg -ffprobe /usr/local/ffmpeg/bin/ffprobe
	Operating system: Unix 3.2.40.0
	Processor count: 2
	64-Bit OS: False
	64-Bit Process: False
	Program data path: /usr/local/emby/var
	Mono: 3.10.0 (tarball Sat Mar 28 10:11:36 UTC 2015)
	Application Path: /volume1/@appstore/emby/share/emby/MediaBrowser.Server.Mono.exe
	Object reference not set to an instance of an object
	System.NullReferenceException
	  at System.Net.NetworkInformation.LinuxNetworkChange.OnAddressChanged (System.Object unused) [0x00000] in <filename unknown>:0 
```

**Comment by LukePulverenti** [comment]
rvmourik - we are using a mono method to listen for network change events. that error is being thrown in a way we can't catch, so for the next release we'll just stop listening to those events.

**Comment by rvmourik** [comment]
@LukePulverenti Okay, thanks.

**Comment by demon1300** [comment]
@cytec any chance we could get the emby packages updated on your repo?

**Comment by cytec** [comment]
@demon1300 im on it... latest version is `3.0.5607.2` right?

**Comment by cytec** [comment]
@Diaoul btw... as emby depends on `sqlite3` and that may take some time to compile, i was thinking about adding `sqlite` directly to the `mono` package... as it's done with python. That would save some time when updating the `emby` package and also provide `sqlite` to other `mono` spk's like `sonarr` for example...

**Comment by demon1300** [comment]
@cytec Yep, that's correct according to their GIT source.

**Comment by cytec** [comment]
SPK's are up

**Comment by demon1300** [comment]
@cytec Awesome work!  Thanks for the quick turn around!

**Comment by LukePulverenti** [comment]
where can the package be found? is it in a state where it can be linked to from the website?

**Comment by cytec** [comment]
@demon1300 strange... should work with those versions as well, i'll check again tomorrow...

@LukePulverenti nope not currently, running but not officially released. I'll let you know when everything is ready for public announcement

**Comment by Vlaves** [comment]
@cytec Any news regarding the patches for ffmpeg and hardware acceleration?

**Comment by Diaoul** [comment]
I had news, Synology already provide the source code, we need to use it to generate a patch and use that patch in spksrc. They won't provide the patch themselves.

**Comment by mortyreKS** [comment]
hello @cytec l have the same bug as @demon1300 with your new version avoton 2.4 (ds1815+) --> it ask new dependencie package. Before I used version x64 1.3 without problem.

**Comment by cytec** [comment]
@mortyreKS @demon1300 uploaded the spk`s again... should work now :)

**Comment by demon1300** [comment]
@cytec Yep, worked without issues!  Thanks again!  If only @LukePulverenti would stop releasing new versions, lol.  It's already outdated!

**Comment by Vlaves** [comment]
@cytec could it be, that for the newest version the evansport build is missing? The x86 build is not installable for me...

**Comment by rvmourik** [comment]
Hi @cytec, the ArmadaXP is also missing i think?

**Comment by LukePulverenti** [comment]
The new stable is tagged on the website. Hoping to see a package we can put on our website soon :)

**Comment by mortyreKS** [comment]
@cytec Nice work !!!  worked without issues! Many Thanks

**Comment by G1zm0** [comment]
@cytec could you update the packages to the latest beta (3.0.5621.2)
I would like to test the improvements (Dramatically Reduced Transcoding CPU Usage)

**Comment by LukePulverenti** [comment]
cytec - Is there anything I can do to help? Any responsibility we can take over to make the packages more easily available? Thanks.

**Comment by cytec** [comment]
@LukePulverenti As far as i remember the only thing that still doesn't work is hardware acceleration support... ImageMagick was just recently merged and should popup on the Repo soon, so we just need to get ffmpeg done and then there shouldn't be any problems publishing emby as well ;)

Just like i said the lack of hardware acceleration support makes it pretty useless for ARM devices if you'll need to transcode something. Is there any option inside Emby to disable transcoding? So that we could Provide an Emby SPK for arm devices but simply disable the transcoding option all together (like Plex Media Server does it) so that the User at least can use Emby to stream to devices which don't require any transcoding?

**Comment by LukePulverenti** [comment]
Yes there is actually. You can disable transcoding at the server level. The client will direct stream if possible, and if not, it will present a friendly error message indicating that the media is unplayable based on configuration.

**Comment by LukePulverenti** [comment]
Although there is not such an option at the package level. But just so we're clear - you would prefer to prevent playback altogether rather than having it make an attempt?

**Comment by cytec** [comment]
well that sounds good to me then... @Diaoul @Dr-Bean what do you think? Personally (like i mentioned somewhere else already) patching ffmpeg for hardware accelleration support would be a lot of work from what i've seen by quickly scrolling through the ffmpeg source synology provides...

@LukePulverenti i _think_ disable playback altogether by default (on arm devices) prevents a lot of "playback is extremely slow/laggy" posts later on... as the hardware is simply not capable of transcoding on those devices. My Personal ideal solution would be something like a check if you'll have enough power to transcode and if not you'll get a message that your hardware doesn't have enough power and therefore transcoding is disabled.

**Comment by cytec** [comment]
just scrolled through this thread http://emby.media/community/index.php?/topic/13890-emby-server-on-synology/ and it seems like i thought... already a lot of questions why it's not playing on arm devices via browser etc...

**Comment by LukePulverenti** [comment]
That thread has a lot of "how to setup" questions. The thread was actually created before you introduced your package. It seems as though the more recent comments are improved, for example:

How to update to the new version, i transform my pc in a nas system now i have a DS3615xs processador intel i install emby and runs very smoth

**Comment by cytec** [comment]
yup saw that as well, but some ppl still asking why it's not correctly working on DS214play (arm) etc

**Comment by Dr-Bean** [comment]
My thoughts: We start off with a beta package. If feedback indicates it's a huge issue, we can always revisit the approach.
We might be able to get transcoding support in our ffmpeg package at some point, or the VideoStation verion can be configured with some manual labor (am I remembering that right?): imo, those are choices we want to leave open at this point.

**Comment by LukePulverenti** [comment]
If you want to disable transcoding by default, I can also add a command line switch for that.

**Comment by cytec** [comment]
@Dr-Bean ok then. so imho only ffmpeg needs to be merged/published and then we are good to go with emby beta as well :)

**Comment by Dr-Bean** [comment]
Time to get rid of the WIP tag on that PR then ;)

**Comment by cytec** [comment]
yeah... sadly i've trashed my VM with the most recent changes/cleanup for that PR, so i'll have to do it all over again and currently i'm lacking a little bit of time... hopefully i'll get to this soon :)

**Comment by The-MAD-Virus** [comment]
hi ! if you have a qoriq version i would be happy to test it for you.

**Comment by dadino** [comment]
I may be looking in the wrong place, but I can't find ImageMagick in https://cytec.us/spk/ and there is no ArmadaXp version in http://cytec.us/syno/test-spk/imagemagick/

So my question is: is it possible to run Emby on a DS414?

**Comment by PeteWhite** [comment]
Hi does anyone know where I can find the 0.7.72-1 version of MediaInfo for the Comcerto CPU?

**Comment by LukePulverenti** [comment]
@cytec just for an update regarding transcoding concerns, we are currently working on a custom video player for android that will be capable of client-side decoding. Once complete, our top clients will all be capable of playback without transcoding, and from there it will just be limited to things like the web interface, roku, windows phone, etc. Hopefully that eases the concerns a little bit, and on top of that we do have settings in the management interface to completely disable it if needed.

**Comment by Gernash** [comment]
Is server side hardware transcoding in the todo list (cuda etc)

**Comment by Vlaves** [comment]
yeah I would also love to have harware transcoding at least on windows if not possible in linux. I would even reinstall my XPEnology system to Windows Home Server for that. Cause there are a lot of low power Intel CPUs, which do have quicksync, so it would make a really good server.

**Comment by Gernash** [comment]
http://emby.media/community/index.php?/topic/10723-gpu-transcoding-intel-quicksync-and-nvidia-nvenc/?p=167070

**Comment by LukePulverenti** [comment]
In addition to everything else above, I thought you guys would also like to know that the current beta eliminates webm transcoding from Chrome and is now fully h264 with HLS. I recall someone raising that as an issue.

**Comment by Vlaves** [comment]
@LukePulverenti good to hear, but that is only relevant if we can have hardware assisted h.264 encoding like on a DS214play or on the quicksync CPUs. So now only a ffmpeg build is missing with the hardware assisted encoding cpapabilities enabled :)

**Comment by Gernash** [comment]
I built and imagicmagick with Q8 and webp seems OK.

**Comment by Gernash** [comment]
But apparently I'm living in the past. Any links to the new mono builds to tryout?

**Comment by cytec** [comment]
@LukePulverenti nice to hear, i think i mentioned it not really as an issue but afaik PMS does all the transcoding with h264 and if i recall correctly transcoding to h264 was faster/more stable then transcoding to webm when i tested it

**Comment by Gernash** [comment]
@cytec
you still using spksrc?
any chance of stealing your folder or uploading it to the git?
or making new version of everything for the syno people :)

**Comment by Gernash** [comment]
@cytec
They say learning is fun but......

**Comment by cytec** [comment]
@Gernash everything there: https://github.com/cytec/spksrc

for emby: https://github.com/cytec/spksrc/tree/emby-server
for ffmpeg: https://github.com/cytec/spksrc/tree/ffmpeg-update

**Comment by Gernash** [comment]
@cytec
It's linux it will always be a work in progress....:) 

Thanks for that, stops me reinventing the wheel. 

1. Are there any people that can be contacted for assistance with specific cross compiling issues that may require patch files(e.g. Mono native config script calling commands that have not been built yet?)

2. ffmpeg- You mentioned PMS earlier in regards to trasncoding can you find out what build switches and ffmpeg version is best/fastest with quality (e.g. was talking to @LukePulverenti and he mentioned cross/libbluray was unused in emby.) or point me in the direction as to who might.

3. Imagemagic - (learnt a lot here) Luke opted for Q8 in the emby server and (before DEV had webm as well) Is there a Guru you know of,  that knows what compile switches are needed for image/transport handling with Emby? (e.g. Need webm and Q8)

The reason for the question is that Synology NAS range from arm to mulit-core home brew servers. So More CPU optimized build for low end processors vs Quality rendering for multicore would be nice.

**Comment by cytec** [comment]
well ffmpeg contains a lot of stuff that isn't used by emby specifically but the more features you have the better... and ffmpeg will be a seperate package which can be shared by more packages than just emby. same goes for imagemagick...

to be honest, most of the stuff is already done in my forks but it still needs cleanup of the make files, plists etc. i've got emby up and running on my Synology for a while now and everything went fine ;)

**Comment by Gernash** [comment]
@cytec 

Same here but it's a bit convoluted atm
http://emby.media/community/index.php?/topic/22921-synology-setup/

If imagemagick Q16 from main SynoCommunity repo is going to be used then imagmagic should be updated/built (some of the libs are newer to) with libwebp Q16 and Q8 (Luke wants Q8 for speed) this would keep everyone happy and not break higher level apps that use it as a dependency.Who would do that for the main Repo? I can happily send them my spksrc Cross and SPK files that do this.

Also there is an issue with ImageMagick and MediaInfo in the evansport and Armada packages.
http://emby.media/community/index.php?/topic/23044-github-repository-for-emby-synology-package/

With those changes Most things will just work out of the Box and your Emby builds can become Live beta and moved to your main repository >.>

**Comment by LukePulverenti** [comment]
Q8/Q16 is up to you guys. I am only recommending Q8 for performance but the server will work with either.

**Comment by cytec** [comment]
@Gernash personally i don't think linking libMagickWand-6.Q8.so to libMagickWand-6.Q16.so  is a good idea... when you'll look at my emby cross folder you'll see i patch emby to use Q16 instead of Q8

https://github.com/cytec/spksrc/blob/emby-server/cross/emby/patches/imagick.patch

alternatively we can update the ImageMagick SPK to build Q8 afaik the only difference is that Q16 is better for "massive manipulations" while Q8 is faster for just resizing and simple stuff like that...

**Comment by G1zm0** [comment]
@cytec can you make an avaton package for ffmpeg?
I'd like to take emby for a spin.

**Comment by Gernash** [comment]
@cytec 

1st thanks for pointing me in the right direction, Have a reasonable grasp of spksrc now.

playing with ffmpeg atm. all I have to say is x264 is driving me crazy can you have a look? Pathing issue?

getting the following
checking for sws_init_context(0,0,0); in libswscale/swscale.h... no
Failed commandline was:
--------------------------------------------------
/home/gernash/spksrc/toolchains/syno-cedarview-5.1/work/x86_64-pc-linux-gnu/bin/x86_64-pc-linux-gnu-gcc conftest.c -m64  -I/home/gernash/spksrc/spk/ffmpeg/work-cedarview-5.1/install//usr/local/ffmpeg/include  -Wall -I. -I$(SRCPATH) -std=gnu99 -mpreferred-stack-boundary=5   -lswscale -lavutil  -m64  -L/home/gernash/spksrc/spk/ffmpeg/work-cedarview-5.1/install//usr/local/ffmpeg/lib -Wl,--rpath-link,/home/gernash/spksrc/spk/ffmpeg/work-cedarview-5.1/install//usr/local/ffmpeg/lib -Wl,--rpath,/usr/local/ffmpeg/lib  -lm -lpthread -o conftest
conftest.c:1:32: fatal error: libswscale/swscale.h: No such file or directory
compilation terminated.
--------------------------------------------------
Failed program was:
--------------------------------------------------
#include <libswscale/swscale.h>
int main (void) { sws_init_context(0,0,0); return 0; }

**Comment by Gernash** [comment]
@cytec 

So cross/libva for ---enable-vaapi (xpenology crew) --enable-vdpau (Synology crew)support is where its at?

So still no idea on last question? cross lib/inclide support/location for compiling x264 without having to compile all the cross dependencies?

**Comment by The-MAD-Virus** [comment]
i'm not sure it's the right place to ask that, 
any way i did manage to compile emby for my DS413 (arch-qoriq) thank's to @cytec.
but i have some trouble with ffmpeg and mono, the first one tell me that "altivec and e500 instructions cannot coexist" and mono tell me is not compatible with my arch.

can some one tell me if is a dead end for qoriq nas ?!

**Comment by cytec** [comment]
@The-MAD-Virus sadly yes, afaik mono does not support qoriq

@Gernash i don't get what you are talking about... x264 and ffmpeg compiles just fine for me...

**Comment by dadino** [comment]
@cytec I can't see ImageMagick in the Community Tab on my DS414. I can see other packages from the community, but not this one. Do I have to do something in particular to get ImageMagick?

**Comment by The-MAD-Virus** [comment]
@cytec thx for the answer so there is still hope ! ( http://www.mono-project.com/docs/about-mono/supported-platforms/powerpc/ )
i well look for it later, but in the meantime is it possible have a working ffmepg for qoriq ? otherwise are this lib necessary or optional ? [flac; libvpx; rtmpdump]

**Comment by LukePulverenti** [comment]
In the latest Emby server release, lipvpx is only used in firefox, and we'll be working to eliminate that. just thought i'd mention that in case it helps. everything else is h264.

**Comment by cytec** [comment]
@dadino strange afaik comcerto2k is available see: https://synocommunity.com/package/imagemagick

@The-MAD-Virus idk if ffmpeg compiles on ppc... haven't tried that... most libs are optional if only used for emby... libvpx is/was needed for transcoding but i think that changed...

**Comment by Gernash** [comment]
@cytec 

MONO O MONO
been bashing my head on that.
Hint for the day m4, automake 1.13 and latest libtool. in usr/local/bin
Especially for 4+ builds.
Banging rocks to make fire.

**Comment by cytec** [comment]
@Gernash why would you even bother with mono? there already is a functional SPK available for mono ;)

**Comment by Gernash** [comment]
@cytec 
@LukePulverenti 

To learn.
Following the CROSS depend chains for ffmpeg, I needed more experience with spksrc.

 I'm not a programmer. I only have basic skills(from the days of UNIX and VAXVMS). Asking for enhancements like adding --enable-vdpau seems to get no response(I assume people don't know so they don't answer). I do not even know if the feature is fully functional in linux as it's only spoke of in dark corners of the internet.
Following the CROSS depend chains for ffmpeg I realized that I needed more experience with spksrc hence the fidddling.

Why am I bothering? What's the point at all, if you OWN a real Synology NAS of using either PLEX or EMBY if transcoding at client level "is not the direction the project is heading" ? HW decoding enabled ffmpeg is needed.

**Comment by LukePulverenti** [comment]
Is there a package available for the latest server version?

**Comment by The-MAD-Virus** [comment]
ok so i manage to compile every thing but i still have some problem -_- .

so can some one tell me the way to determine if it's mono or something else ?
reminder: i use a DS413 with qoriq cpu.
```shell
=================================================================
Got a SIGABRT while executing native code. This usually indicates
a fatal error in the mono runtime or one of the native libraries
used by your application.
=================================================================
```

**Comment by sorcer1** [comment]
Like @LukePulverenti, it wiil fantastic if the server can be packaged again with the latest version.

**Comment by Vlaves** [comment]
Yeah would be great to have an updated version. If I update the current Version with WinSCP it is not working properly. @LukePulverenti: Would it then be possible to host the relevant packages on the main emby site, with all the dependent packages?

**Comment by techabyte** [comment]
Would love this to be running on my nas!

**Comment by LukePulverenti** [comment]
cytec - any chance you can do an update to the latest version? thanks.

**Comment by cytec** [comment]
i've lost my vm on which i was mostly working on this...
I'm gonna setup a new one once i have some free time ...
Hopefully starting next week ;)

**Comment by LukePulverenti** [comment]
Thanks. Were you able to build it?

**Comment by cytec** [comment]
@LukePulverenti yup... most recent version is `5.1_3.0.5724.5` right? currently compiling it and then uploading it to my server... 

I've also updated the PR and changed the Logo, let's hope the PR will be merged soon ;)

**Comment by LukePulverenti** [comment]
Where's the PR, just so that I can track it? thanks.

**Comment by cytec** [comment]
PR can be found here: https://github.com/SynoCommunity/spksrc/pull/1673

**Comment by cytec** [comment]
@LukePulverenti currently uploading the new version so they should pop up soon ;) 
And sorry again it took me so long to update it -.-

**Comment by LukePulverenti** [comment]
thanks, really appreciate it. Can you do one more thing? can you tell if if these installation instructions are still valid for the new package?

http://emby.media/community/index.php?/topic/22921-synology-setup/

Thanks!

**Comment by cytec** [comment]
yup, nothing changed, just the version of emby :)

**Comment by LukePulverenti** [comment]
are the manual dependency installations required, or does the package handle that by marking it's dependencies?

**Comment by cytec** [comment]
ffmpeg and emby must both be downloaded as stated in your guide... 
if you try to install emby before installing the depends, it'll show a message that you need to install some dependencies. and yes dependencies must always be installed by the user manually (afaik)

**Comment by LukePulverenti** [comment]
cytec, what do you make of this statement?

The latest armadaxp build at http://cytec.us/syno/test-spk/emby/ seems broken. It's way smaller than the other files and my diskstation shows an error message when trying to install it ("invalid file format")

http://emby.media/community/index.php?/topic/22921-synology-setup/page-6#entry255352

**Comment by cytec** [comment]
@LukePulverenti seems like that one wasn't uploaded correctly. i re-uploaded it so it should work now :)

**Comment by LukePulverenti** [comment]
Thanks!

**Comment by G1zm0** [comment]
I want to say I have the latest server running on om my DS1815+ for a few months now.
A few remarks:
- restart is not working (no problem, just use the package centre)
- Updating is manually, so a copy-paste of the new source.

I've been so pleased, I bought the premiere. Espcially with the latest iOS app release.
So this is a thumbs up for @cytec en I hope the PR gets pulled soon.

**Comment by LukePulverenti** [comment]
Thanks for the kind words! Hopefully it gets merged soon. In fact, maybe that's where our comments should be directed now?

https://github.com/SynoCommunity/spksrc/pull/1673

**Comment by cytec** [comment]
small update: hopefully the ffmpeg PR get's merged soon *fingers crossed* @LukePulverenti does emby provide a native updater? basically what i want to know is if emby is able to update itself? that would be great because we don't have to build new SPK's everytime a new version is out :/

**Comment by LukePulverenti** [comment]
At the moment we only have a native updater for windows but considering the challenge we've had in getting packages released and updated it's definitely something we should look into.

**Comment by cytec** [comment]
@LukePulverenti yeah for synology it would be as simple as downloading the zip from the repo and extract the files to a specific dir :) That would be a LOT faster/easyer then recompiling the SPK everytime as it needs busybox and sqlite support (which is build on SPK creation)... or maybe there is another way to automize that... so that it basically auto generates a spk if a new version is pushed to github... that may be an option too :)

**Comment by LukePulverenti** [comment]
I am perfectly fine with building the native update, it's really no problem at all. My main concern right now is just getting into the Synology catalog for easy installation, because the manual installation can be painful.

**Comment by cytec** [comment]
yup i'm with you on that one ;)

**Comment by abobader** [comment]
I hope this be done in the near future.

**Comment by schumi2004** [comment]
I think this issue can be closed since Emby already has it's own repo and package.

**Comment by cytec** [comment]
yup
